packages/eslint-plugin/tests/rules/member-ordering.test.ts

```diff
@@ -3979,6 +3999,57 @@ class Foo {
         },
       ],
     },
+    {
+      code: `
+class Foo {
+  static {}
+  m() {}
+  f = 1;
+}
+            `,
+      options: [{ default: ['method', 'field', 'static-initialization'] }],
+      errors: [
+        {
+          messageId: 'incorrectGroupOrder',
+          data: {
+            name: 'm',
+            rank: 'static initialization',
+          },
+          line: 4,
+          column: 3,
+        },
+        {
+          messageId: 'incorrectGroupOrder',
+          data: {
+            name: 'f',
+            rank: 'static initialization',
+          },
+          line: 5,
+          column: 3,
+        },
+      ],
+    },
+    {
+      code: `
+class Foo {
+  m() {}
+  f = 1;
+  static {}
```

Comment:
Could you also add test cases for `static {}` being intermixed with other blocks? Just to make sure there are no funky edge cases in ordering behavior.

Reply:
By continuity principle (something I just made up) the tests should be sufficient. Let me know if there are specific quirks worth testing that you have in mind.